### ISSUE TITLE
HuBERT training 

### ISSUE THREAD

**Issue by gustmd0121** [facebookresearch/fairseq#4312]
### Before asking:
1. search the issues.
2. search the docs.

#### What is your question?
Training loss for HuBERT 
While pre-training for HuBERT I've encountered the following changes in loss (red: validation) 
I was wondering if this is just a case of overfitting, and if anyone else encountered this issue?
Thank you. 
#### Code
![voerfit](https://user-images.githubusercontent.com/49477579/160273509-64f47779-4870-41e3-afd1-23da13a5c169.PNG)

#### What's your environment?

 - fairseq Version (e.g., 1.0 or main):
 - PyTorch Version (e.g., 1.0)
 - OS (e.g., Linux):
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by wangtianrui** [comment]
I also encountered this problem. In the end, I found out that I made a very low-level mistake. I trained a separate km model to generate labels based on the validation set and the training set, which resulted in a mismatch between the labels of the training set and the labels of the validation set. Although this does not affect the final effect of the model (just pick the last saved model), this problem still needs to be avoided.

**Comment by SevKod** [comment]
I had the same issue. The README.md from HuBERT pretraining has confusing parts. The "learn_kmeans.py" script has to be applied only to the TRAIN set (and not the ${split} which is both the train and valid).

This issue happens when you train a kmean model (using "learn_kmeans.py") on both the train and validation set , which makes your valid and train clusters completely different. This results  in a complete mismatch (between clusters) when computing Validation Loss. 
The clusters from the train and validation sets have to be the SAME. So, you should compute the "learn_kmeans.py" only on the TRAIN set, then use "dump_km_label.py" on both the train and validation using the kmean-model generated from the training loss, using "learn_kmeans.py".
